pkg/storage/batch.go

```diff
@@ -22,6 +24,55 @@ import (
 	"github.com/grafana/loki/pkg/logql/stats"
 )
 
+type ChunkMetrics struct {
+	refs         prometheus.Counter
+	filteredRefs prometheus.Counter
+	chunks       *prometheus.CounterVec
+	batches      prometheus.Histogram
+}
+
+const (
+	statusFiltered = "filtered"
+	statusMatched  = "matched"
+)
+
+func NewChunkMetrics(r prometheus.Registerer, maxBatchSize int) *ChunkMetrics {
+	buckets := 5
+	if maxBatchSize < buckets {
+		maxBatchSize = buckets
+	}
+
+	return &ChunkMetrics{
+		refs: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "chunk_refs_downloaded_total",
+			Help:      "Number of chunks refs downloaded.",
+		}),
+		filteredRefs: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "chunk_refs_filtered_downloaded_total",
+			Help:      "Number of chunks refs downloaded whose bounds intersect the query bounds.",
+		}),
+		chunks: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "chunks_downloaded_total",
+			Help:      "Number of chunks downloaded, partitioned by if they satisfy matchers.",
+		}, []string{"status"}),
+		batches: promauto.With(r).NewHistogram(prometheus.HistogramOpts{
+			Namespace: "loki",
+			Subsystem: "store",
+			Name:      "effective_batch_size",
```

Comment:
Should we prefix the name with `chunks` or maybe call it `chunks_per_batch_post_filtering` ?